### PR TITLE
Spring4 & integration tests with Docker

### PR DESCRIPTION
**Pull request by djtfmartin**
Ive started creating integration tests that use Docker instances in the travis build.
Further work is to maybe use the fabric8 maven plugin for running docker instances.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,3 +11,14 @@ There are currently two supported versions:
...
 * 2.x - SOLR 7 with SOLR Cloud support and Cassandra 3.x. See the master branch.
+
+## Integration Tests
+
+Integration testing is supported using docker containers.
+To start the required containers, run the following:
+
+```
+docker-compose -f src/test/docker/solr7-cassandra3.yml up -d
+```
```
Comment (ansell) [inline README.md]: This also needs the instruction from .travis.yml for killing the docker-compose after finished working with it:

```
docker-compose -f src/test/docker/solr7-cassandra3.yml kill
```

File: src/main/java/au/org/ala/biocache/service/AuthService.java [hunk 2]
```diff
@@ -14,6 +14,7 @@
  ***************************************************************************/
 package au.org.ala.biocache.service;
 
+import com.drew.lang.StringUtil;
```
[the remark this answers is not in the document]
Comment (djtfmartin) [reply src/main/java/au/org/ala/biocache/service/AuthService.java]: good catched. Removed

File: README.md (hunk 1 above)
Comment (djtfmartin) [reply README.md]: good idea. Added

**Comment by djtfmartin** [comment]
Thanks @ansell. Im going to leave the cassandra and SOLR data files for now. Im not concerned with unresolvable conflicts for these binary files. If the are changed then it means the docker images that the tests use will need to be changed. The files arent used by the tests as the files are included in the images that are downloaded by docker. I've left there here so that the docker images can be regenerated more easily.

**Pull request merged**